resource/abstractions/auditing_rules.md

```diff
@@ -0,0 +1,478 @@
+  The type of **attempt** (Success/Failure/All) for which read access shall be audited. Ignore this operand to specify a different "access level".
+
+* `update`<br>
+  The type of **attempt** (Success/Failure/All) for which update access shall be audited. Ignore this operand to specify a different "access level".
+
+#### 📤 Returns
+* `Union[dict, bytes]`<br>
+  Returns a **Security Result Steps dictionary** or **Concatenated Security Request XML bytes** if the `ResourceAdmin.generate_requests_only` class attribute is set to `True`.
+
+#### ❌ Raises
+* `SecurityRequestError`<br>
+  Raises `SecurityRequestError` when the **SAF Return Code** of a **Security Result** is equal to `4`.
+* `DownstreamFatalError`<br>
+  Raises `DownstreamFatalError` when the **SAF Return Code** of a **Security Result** is greater than `4`.
+* `AlterOperationError`<br>
+  Raises `AlterOperationError` when the **general resource profile** cannot be altered because it does **NOT** exist.
+
+#### 💻 Example
+
+###### Python REPL
+```python
+>>> from pyracf import ResourceAdmin
+>>> resource_admin = ResourceAdmin()
+>>> resource_admin.overwrite_audit_by_access_leve("TESTING","ELIJTEST",alter="success")
+{"step1":{"securityResult":{"resource":{"name":"TESTING","class":"ELIJTEST","operation":"set","requestId":"ResourceRequest","info":["Definition exists. Add command skipped due  to precheck option"],"commands":[{"safReturnCode":0,"returnCode":0,"reasonCode":0,"image":"RALTER  ELIJTEST             (TESTING)  AUDIT( success (ALTER       ))","messages":["ICH11009I RACLISTED PROFILES FOR ELIJTEST WILL NOT REFLECT THE UPDATE(S) UNTIL A SETROPTS REFRESH IS ISSUED."]}]},"returnCode":0,"reasonCode":0}}}
+```
+
+###### Security Result Steps Dictionary as JSON
+```json
+{
+  "step1": {
+    "securityResult": {
+      "resource": {
+        "name": "TESTING",
+        "class": "ELIJTEST",
+        "operation": "set",
+        "requestId": "ResourceRequest",
+        "info": [
+          "Definition exists. Add command skipped due  to precheck option"
+        ],
+        "commands": [
+          {
+            "safReturnCode": 0,
+            "returnCode": 0,
+            "reasonCode": 0,
+            "image": "RALTER  ELIJTEST             (TESTING)  AUDIT( success (ALTER       ))",
+            "messages": [
+              "ICH11009I RACLISTED PROFILES FOR ELIJTEST WILL NOT REFLECT THE UPDATE(S) UNTIL A SETROPTS REFRESH IS ISSUED."
+            ]
+          }
+        ]
+      },
+      "returnCode": 0,
+      "reasonCode": 0
+    }
+  }
+}
+```
+
+## `ResourceAdmin.alter_audit_by_attempt()`
+
+```python
+def alter_audit_by_attempt(
+    self,
+    resource: str,
+    class_name: str,
+    success: Union[str, None] = None,
+    failure: Union[str, None] = None,
+    all: Union[str, None] = None,
+) -> Union[dict, bytes]:
+```
+
+#### 📄 Description
+
+Preserves any rules defined within the **Audit Rules** for a general resource profile and add new `Success`, `Failure`, or `All` rule(s).
```

Comment:
Alter the **Auditing Rules** of a general resource profile specified by **Access Attempt**.

Reply:
Change made and applied to other Alter function.